gdal/frmts/sigdem/sigdemdataset.h

```diff
@@ -0,0 +1,153 @@
+/******************************************************************************
+ *
+ * Project:  Scaled Integer Gridded DEM .sigdem Driver
+ * Purpose:  Implementation of Scaled Integer Gridded DEM
+ * Author:   Paul Austin, paul.austin@revolsys.com
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Paul Austin <paul.austin@revolsys.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#ifndef GDAL_FRMTS_SIGDEMDATASET_H_INCLUDED
+#define GDAL_FRMTS_SIGDEMDATASET_H_INCLUDED
+
+#include "cpl_port.h"
+
+#include <cctype>
+#include <cerrno>
+#include <climits>
+#include <cmath>
+#include <cstddef>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#if HAVE_FCNTL_H
+#  include <fcntl.h>
+#endif
+
+#include <limits>
+#include <memory>
+
+#include "cpl_conv.h"
+#include "cpl_error.h"
+#include "cpl_progress.h"
+#include "cpl_string.h"
+#include "cpl_vsi.h"
+#include "gdal.h"
+#include "gdal_frmts.h"
+#include "gdal_pam.h"
+#include "gdal_priv.h"
+#include "gdal_rat.h"
+#include "ogr_core.h"
+#include "ogr_spatialref.h"
+
+class SIGDEMRasterBand;
+
+#pragma pack(push, 1)
```

Comment:
I'm not a big fan of this. It might not be honoured by all compilers and could cause issues on architectures where this doesn't match their alignment requirements